EIPS/eip-6358.md

```diff
@@ -12,37 +12,37 @@ created: 2023-01-17
-- **Synchronizer** is an off-chain procedure responsible for synchronizing published omniverse transactions from the O-DLT smart contracts on one blockchain to the O-DLT smart contracts of the others.
+With this ERC, we can create a global token protocol, that leverages smart contracts or similar mechanisms on existing blockchains to record the token states synchronously. The synchronization could be made by trustless off-chain synchronizers.
 
 ## Motivation
 
-For projects serving multiple chains, it might be useful that the token is able to be accessed anywhere. Although assets-bridges can more or less make it, we don't think it is enough. And in the process of R&D, we found that the fragmentation of tokens is a common and disturbing problem among chains and L2s. 
+- The current paradigm of token bridges makes assets fragment.  
+- If ETH was transferred to another chain through the current token bridge, if the chain broke down, ETH will be lost for users.  
 
-- We want our token to be treated as a whole instead of being divided into fragmented parts on different public chains. O-DLT can get it.
-- When one chain breaks down, we don't want to lose our assets along with it. Assets-bridge paradigm cannot provide a guarantee for this. O-DLT can provide this guarantee even if there's only one chain that works, for example, we can rely on the stability and robustness of Ethereum.  
-- Not just for a certain token, we think the Omniverse Token might be useful for other projects on Ethereum and other chains. O-DLT is actually a new kind of open-source asset paradigm at the application level. 
+The core of this ERC is synchronization instead of transferring, even if all the other chains break down, as long as Ethereum is still running, user’s assets will not be lost. 
 
+- The fragment problem will be solved.
+- The security of users' multi-chain assets can be greatly enhanced.
 
 ## Specification
 
 The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
 
-*The word `Omniverse` in definitions will be substituted with the assigned number.*
-
 ### Omniverse Account
 
-The Omniverse account is RECOMMENDED to be expressed as a public key created by the elliptic curve `secp256k1`, which has already been supported by Ethereum tech stacks and can be used to generate an Ethereum address directly. For those who have a different address system, a mapping mechanism is RECOMMENDED.  
+There SHOULD be a global user identifier of this ERC, which is RECOMMENDED to be referred to as Omniverse Account (`o-account` for short) in this article.  
+The `o-account` is RECOMMENDED to be expressed as a public key created by the elliptic curve `secp256k1`. A mapping mechanism is RECOMMENDED for different environments.  
```

Comment:
Could you elaborate on what a mapping mechanism is?

Reply:
In the simplest implementation, we can just build two mappings to get it. One is like `pk based on sece256k1 => account address in the special environment`, and the other is the reverse mapping. 

The [Account System](https://developers.flow.com/tools/flow-js-testing/accounts) on [Flow](https://flow.com/) is a typical example. We have provided an implementation example [here](https://github.com/Omniverse-Web3-Labs/omniverse-flow)

- The public key can be bound to an account (a special built-in data structure) and the public key can be got from the `account address` like [this](https://github.com/Omniverse-Web3-Labs/omniverse-flow/blob/ee68e77200e8f804b815b4fa634096e90c3b1a89/contracts/OmniverseProtocol.cdc#L433)
- A mapping from `pk from sece256k1` to the `account address` on Flow can be built like [this](https://github.com/Omniverse-Web3-Labs/omniverse-flow/blob/ee68e77200e8f804b815b4fa634096e90c3b1a89/contracts/OmniverseProtocol.cdc#L6) and we can get the `account address` from `pk from sece256k1` like [this](https://github.com/Omniverse-Web3-Labs/omniverse-flow/blob/ee68e77200e8f804b815b4fa634096e90c3b1a89/contracts/OmniverseProtocol.cdc#L507)

Reply:
In the proposal itself please!

Reply:
Hi @SamWilsn ,

Sincerely thank you for your patience and suggestions, and we have added the related contents to the proposal [here](https://github.com/ethereum/EIPs/blob/dfbf00d976d92115165a5be1185db12d8e4db255/EIPS/eip-6358.md#omniverse-account) and [here](https://github.com/ethereum/EIPs/blob/dfbf00d976d92115165a5be1185db12d8e4db255/EIPS/eip-6358.md#mapping-mechanism-for-different-environments).

All the Best~